apps/studio/components/interfaces/BranchManagement/Overview.tsx

```diff
@@ -308,12 +313,20 @@ const PreviewBranchActions = ({
             className="gap-x-2"
             disabled={!canDeleteBranches}
             onSelect={(e) => {
-              e.stopPropagation()
-              onSelectDeleteBranch()
+              if (isPersistentBranch) {
+                setShowPersistentBranchDeleteConfirmationModal(true)
+              } else {
+                e.stopPropagation()
+                onSelectDeleteBranch()
+              }
             }}
             onClick={(e) => {
-              e.stopPropagation()
-              onSelectDeleteBranch()
+              if (isPersistentBranch) {
+                setShowPersistentBranchDeleteConfirmationModal(true)
+              } else {
+                e.stopPropagation()
+                onSelectDeleteBranch()
```

Comment:
Shall we move this check inside `onSelectDeleteBranch` to reduce duplication?

Reply:
Can't put it in there since it comes from a props. But I can make a wrapper around this to dedup 👍